findex_in_memory/src/findex.rs

```diff
@@ -0,0 +1,248 @@
+        Ok(rejected)
+    }
+
+    async fn insert_chain_table(
+        &mut self,
+        items: &EncryptedTable<UID_LENGTH>,
+    ) -> Result<(), Error> {
+        //println!("Insert {} items in the Chain Table", items.len());
+        for (uid, value) in items.iter() {
+            if self.chain_table.contains_key(uid) {
+                return Err(Error::CallBack(format!(
+                    "Conflict in Chain Table for UID: {uid:?}"
+                )));
+            }
+            self.chain_table.insert(uid.clone(), value.clone());
+        }
+        Ok(())
+    }
+
+    fn update_lines(
+        &mut self,
+        chain_table_uids_to_remove: HashSet<Uid<UID_LENGTH>>,
+        new_encrypted_entry_table_items: EncryptedTable<UID_LENGTH>,
+        new_encrypted_chain_table_items: EncryptedTable<UID_LENGTH>,
+    ) -> Result<(), Error> {
+        //println!(
+        //"Remove {} items from the Chain Table",
+        //chain_table_uids_to_remove.len()
+        //);
+        //println!(
+        //"Insert {} items to the Chain Table",
+        //new_encrypted_chain_table_items.len()
+        //);
+        //println!(
+        //"Insert {} items to the Entry Table",
+        //new_encrypted_entry_table_items.len()
+        //);
+
+        self.entry_table = EncryptedTable::default();
+
+        for new_encrypted_entry_table_item in new_encrypted_entry_table_items.iter() {
+            self.entry_table.insert(
+                new_encrypted_entry_table_item.0.clone(),
+                new_encrypted_entry_table_item.1.clone(),
+            );
+        }
+
+        for new_encrypted_chain_table_item in new_encrypted_chain_table_items.iter() {
+            self.chain_table.insert(
+                new_encrypted_chain_table_item.0.clone(),
+                new_encrypted_chain_table_item.1.clone(),
+            );
+        }
+
+        for removed_chain_table_uid in chain_table_uids_to_remove {
+            self.chain_table.remove(&removed_chain_table_uid);
+        }
+
+        Ok(())
+    }
+
+    fn list_removed_locations(&self, _: &HashSet<Location>) -> Result<HashSet<Location>, Error> {
+        Ok(self.removed_locations.iter().cloned().collect())
+    }
+
+    async fn fetch_all_entry_table_uids(&self) -> Result<HashSet<Uid<UID_LENGTH>>, Error> {
+        let uids: HashSet<Uid<UID_LENGTH>> = self.entry_table.keys().cloned().collect();
+        Ok(uids)
+    }
+}
+
+impl
+    FindexSearch<
+        UID_LENGTH,
+        BLOCK_LENGTH,
+        TABLE_WIDTH,
+        MASTER_KEY_LENGTH,
+        KWI_LENGTH,
+        KMAC_KEY_LENGTH,
+        DEM_KEY_LENGTH,
+        KmacKey,
+        DemScheme,
+        Error,
+    > for FindexInMemory<UID_LENGTH>
+{
+}
+
+impl
+    FindexUpsert<
+        UID_LENGTH,
+        BLOCK_LENGTH,
+        TABLE_WIDTH,
+        MASTER_KEY_LENGTH,
+        KWI_LENGTH,
+        KMAC_KEY_LENGTH,
+        DEM_KEY_LENGTH,
+        KmacKey,
+        DemScheme,
+        Error,
+    > for FindexInMemory<UID_LENGTH>
```

Comment:
We should create a type alias for this… Maybe not in this PR…

Reply:
I agree, but I prefer this to be done in a latter PR :pray:

Reply:
Done in #25